### ISSUE TITLE
renew parallel/async renew

### ISSUE THREAD

**Issue by alphaonex86** [certbot/certbot#3330]
Hello,

I need shutdown my web server to start this command:
letsencrypt renew --email "alpha_one_x86@first-world.info" --agree-tos --rsa-key-size 4096
But the problem it's, I have 31 sub domain into /etc/letsencrypt/live/, I don't wish change my structure. It's at least minutes of down due to serial pooling. Can you speed up doing this query in parallel/async to query multiple certificate at times, and prevent round-trip delay time problem?

Cheers,

**Comment by bmw** [comment]
I'm assuming you're using the standalone plugin? We can definitely look into something like this as it would be a useful feature, however, it'll be tricky on our end.

In the meantime, does the [webroot plugin](https://certbot.eff.org/docs/using.html#webroot) not work for your use case? If your server is serving files at a specified path (such as `/var/www/html` or `/usr/share/nginx/html`), `webroot` allows you to complete the challenges with no server downtime.

For those interested on why this will be tricky, having different plugin instances trying to do challenges in parallel isn't likely to work well (this is true whether the plugins have the same or different types). Furthermore, attempting to do the challenges for multiple lineages at the same time becomes complex as values in `/etc/letsencrypt/renewal` conflict. Perhaps we can do this only if the renewal params in `/etc/letsencrypt/renewal` are the same? This is much less valuable though.

**Comment by alphaonex86** [comment]
Yes I'm using the standalone plugin, the other plugins was not working when I have test.

But I use gentoo, and not standard config.

Here only the domain change, but maybe later I will have another params.

**Comment by bmw** [comment]
I'd give the `webroot` plugin a try again and see if it works for you. If your server responds to HTTP requests and you're serving HTML files at a certain path, `webroot` should work. You simply give it the path where to place files that will be served by your web server. It does not attempt to read/parse/modify your server's config at all.

Even if that works for you, we should leave this issue open as it'd be a useful feature.

**Comment by alphaonex86** [comment]
Remain waiting of this feature

**Comment by alphaonex86** [comment]
Hi, remain valid

**Comment by alphaonex86** [comment]
You don't known if your script able to do multi cert in parallel

**Comment by osirisinferi** [comment]
> But I use gentoo, and not standard config.

I'm using Gentoo too and I'm *pretty* sure using `standalone` is not necessary. Maybe you could open a thread in our [Let's Encrypt Community](https://community.letsencrypt.org/) so we could look into your specific situation? As parallel renewal might not be a feature happening anytime soon.

**Comment by bmw** [comment]
I doubt we're going to do this, especially anytime soon. It'd be a major overhaul for us and there isn't much demand for it.
